### PR TITLE
7.6/offline mode

### PR DESCRIPTION
**Pull request by lguardado**
#### :page_facing_up: Description:

Add offline mode.
The app should keep the date when the app is online. When internet is back, it should keep fetching new data normally.
The check happens when the app is opened and/or when the app is unfocused and focused again.
The user can now set the expiration date of the data so when it gets past due it's re-fetched or deleted (if no internet connection).

---

#### :pushpin: Notes:

- In the demo there is an alert showing up to show we're using the movies titles to trigger and old data and re-fetching the movies. When you go back to home, the movies are fetch already.

---

#### :heavy_check_mark: Performed Tasks:

- Add offline mode.
- Add config for picking the data expiration
- Add tests
- Add placeholder images for when there is no image fetched / offline mode.

---

#### :play_or_pause_button: Demo:
Kapture 2020-12-07 at 16 29 21
Kapture 2020-12-07 at 16 31 41
Kapture 2020-12-07 at 16 42 50
Screen Shot 2020-12-07 at 16 54 29
Screen Shot 2020-12-07 at 16 56 25

### CODE REVIEW

File: src/components/Details/styles.js [hunk 1]
```diff
@@ -10,6 +10,15 @@ const styles = StyleSheet.create({
   movieCard: {
     height: 300,
   },
+  movieCardPlaceholder: {
+    backgroundColor: Colors.lightGray,
+    alignItems: 'center',
+    width: '100%',
```
Comment (Jmaguerre) [inline src/components/Details/styles.js:16]: Is it necessary to have this and flex 1?

File: src/components/Home/Home.js [hunk 2]
```diff
@@ -41,12 +58,83 @@ function Home({ navigation }) {
...
+    },
+    [dispatch, movies]
+  );
+
+  const handleAppStateChange = useCallback(
+    nextAppState => {
+      if (
+        appState.current.match(/inactive|background/) &&
+        nextAppState === 'active'
+      ) {
```
Comment (Jmaguerre) [inline src/components/Home/Home.js:99]: This is a weird syntax for an if, did the linter suggest this?

**Review by Jmaguerre — APPROVED** [review]
Great job! Some minor things

File: src/components/Home/Home.js (hunk 2 above)
Comment (lguardado) [reply src/components/Home/Home.js:99]: linter 😂

File: src/components/Details/styles.js (hunk 1 above)
Comment (lguardado) [reply src/components/Details/styles.js:16]: none of them is necessary actually! removing...

**Review by Jmaguerre — APPROVED** [review]

**Pull request merged**